react-app/src/modules/account/account.test.js

```diff
@@ -0,0 +1,25 @@
+import * as account from './account.js';
+
+afterEach(() => account.clearAccount());
+
+test('creates keypairs from username and password', async () => {
+  expect.assertions(4);
+  await account.constructor('$/()RUFJ)wuwe84349', 'dnefu384(/·$873hf7g')
+    .then(result => {
+      expect(result).toBeNull();
+      expect(account.getUsername()).toBe('$/()RUFJ)wuwe84349');
+      expect(btoa(String.fromCharCode.apply(null, account.getPublicKey())))
+        .toBe('6FUygceUHlKzGu/5ir4FNkmAtUF07uNeAAdp9+5hcy4=');
+      expect(btoa(String.fromCharCode.apply(null, account.getSecretKey())))
+        .toBe('m4jBZuDiCgcCz94VIWHmeVC9IsXOIavniA2pqeq5Gg0=');
+    });
+});
+
+test('encrypt and decrypt a message', () => {
+  expect.assertions(2);
+  account.setKeyPair('6FUygceUHlKzGu/5ir4FNkmAtUF07uNeAAdp9+5hcy4=',
+                     'm4jBZuDiCgcCz94VIWHmeVC9IsXOIavniA2pqeq5Gg0=');
+  let encryptedMessage = account.encrypt('sómething with ñ and ü\n!_¡?\'{[^]}}à');
+  expect(typeof encryptedMessage).toBe('string');
+  expect(account.decrypt(encryptedMessage)).toBe('sómething with ñ and ü\n!_¡?\'{[^]}}à');
```

Comment:
You should add a test where you modify what you encrypted and it fails at decryption.

Reply:
The problem I face if I change the encrypted text is that I get and invalid base64 encoded cipherText for decrypt, so I get an wrong encoding error, that if OK, is not what I want to test.

Maybe I misunderstood the expected output and it should be UTF8 encoded for local use and only be encoded to base64 before sending it to the server.

Reply:
Decode -> modify -> encode should work. Upload the failing test, maybe having more eyes looking at it would make the bug clear.